symfony/plugins/orangehrmAttendancePlugin/Controller/EditAttendanceController.php

```diff
@@ -20,28 +20,42 @@
 
 namespace OrangeHRM\Attendance\Controller;
 
+use OrangeHRM\Attendance\Traits\Service\AttendanceServiceTrait;
 use OrangeHRM\Core\Controller\AbstractVueController;
+use OrangeHRM\Core\Controller\Common\NoRecordsFoundController;
+use OrangeHRM\Core\Controller\Exception\RequestForwardableException;
 use OrangeHRM\Core\Vue\Component;
 use OrangeHRM\Core\Vue\Prop;
+use OrangeHRM\Entity\AttendanceRecord;
 use OrangeHRM\Framework\Http\Request;
-use OrangeHRM\Core\Controller\Common\NoRecordsFoundController;
-use OrangeHRM\Core\Controller\Exception\RequestForwardableException;
 
 class EditAttendanceController extends AbstractVueController
 {
+    use AttendanceServiceTrait;
+
     /**
      * @inheritDoc
      */
     public function preRender(Request $request): void
     {
         if ($request->attributes->has('id')) {
+            $attendanceRecordId = $request->attributes->getInt('id');
+            $attendanceRecord = $this->getAttendanceService()
+                ->getAttendanceDao()
+                ->getAttendanceRecordById($attendanceRecordId);
+            //no attendance record for the given id
+            if (!$attendanceRecord instanceof AttendanceRecord) {
+                throw new RequestForwardableException(NoRecordsFoundController::class . '::handle');
+            }
             $component = new Component('edit-attendance');
-            $component->addProp(new Prop('attendance-id', Prop::TYPE_NUMBER, $request->attributes->getInt('id')));
+            //check auth user's permission to update attendance record
+            if ($this->getAttendanceService()->isAuthUserAllowedToPerformTheEditActions($attendanceRecord)) {
+                $component->addProp(new Prop('is-editable', Prop::TYPE_BOOLEAN, true));
```

Comment:
@SandeepaOHRM I need to block user from here if they are not allowed. since this screen is a edit screen no point passing a prop.

Reply:
fixed #[7353113](https://github.com/orangehrm/orangehrm/pull/1135/commits/7353113283824d3793e07e3b1e1ab3eb61e8e55f)